lekko_client/clients/distribution_client.py

```diff
@@ -67,10 +74,11 @@ def _upload_events(self) -> None:
                 self.events = []
 
         def _accept_event(self) -> None:
-            try:
-                self.events.append(self.queue.get(block=False))
-            except queue.Empty:
-                pass
+            # Block until an event is ready
+            event = self.queue.get()
+            if event is None:  # Termination signal
```

Comment:
nit: if you make `_accept_event` return `bool` then we don't need `_enabled`, we can run until `_accept_event` returns `False`.

Reply:
Sounds good, that's a lot cleaner.